### PR TITLE
Replace dict with style parameters

### PR DESCRIPTION
**Pull request by Yun-Wu**
- **Define layer styles**
- **ecomap**
- **test with mode-map**
- **handle defs**
- **ecoplot**
- **patrol workflow**
- **mapvalues**

### CODE REVIEW

File: ecoscope_workflows/tasks/results/_ecomap.py [hunk 1]
```diff
@@ -1,55 +1,94 @@
...
+
+
+LayerStyle = Annotated[
+    Union[PolylineLayerStyle, PointLayerStyle, PolygonLayerStyle],
+    Field(discriminator="layer_type"),
+]
+
+
+class NorthArrowStyle(BaseModel):
+    placement: Literal[
```
Comment (walljcg) [inline ecoscope_workflows/tasks/results/_ecomap.py:48]: Is it possible we use relative coords for this? These kws don't give us quite enough control over where to position the arrow...

Comment (Yun-Wu) [reply ecoscope_workflows/tasks/results/_ecomap.py:48]: I got the list from here:

https://github.com/wildlife-dynamics/lonboard/blob/main/lonboard/_deck_widget.py#L7

@atmorling do you know if it supports relative coords? Or should we just implement it in ecoscope-core?

Comment (atmorling) [reply ecoscope_workflows/tasks/results/_ecomap.py:48]: You can do it currently via a combination of style kws. I've raised an [issue](https://github.com/wildlife-dynamics/ecoscope/issues/246) on `ecoscope` as I think it makes sense to wrap it there.

File: ecoscope_workflows/tasks/results/_ecoplot.py [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+
+class GroupedPlotStyle(BaseModel):
+    category: str
+    plot_style: PlotCategoryStyle
+
+
+class LayoutStyle(BaseModel):
+    font_color: str | SkipJsonSchema[None] = None
+    font_style: str | SkipJsonSchema[None] = None
+
```
Comment (atmorling) [inline ecoscope_workflows/tasks/results/_ecoplot.py:33]:
```suggestion
class LayoutStyle(BaseModel):
    font_color: str | SkipJsonSchema[None] = None
    font_style: str | SkipJsonSchema[None] = None
    plot_bgcolor: str | SkipJsonSchema[None] = None
    showlegend: bool | SkipJsonSchema[None] = None


class BarLayoutStyle(LayoutStyle):
    bargap: Annotated[float, ValueRange(0.0, 1.0)] | SkipJsonSchema[None] = None
    bargroupgap: Annotated[float, ValueRange(0.0, 1.0)] | SkipJsonSchema[None] = None
```
Just some extra style params that may be useful

File: ecoscope_workflows/tasks/results/_ecoplot.py [hunk 3]
```diff
@@ -82,21 +107,19 @@ def draw_time_series_bar_chart(
...
-        dict | SkipJsonSchema[None],
-        Field(
-            description="Style arguments passed to plotly.graph_objects.Bar and applied to all groups."
-        ),
-    ] = None,
-    layout_kws: Annotated[
-        dict | SkipJsonSchema[None],
-        Field(description="Style arguments passed to plotly.graph_objects.Figure."),
+    layout_style: Annotated[
+        LayoutStyle | SkipJsonSchema[None],
```
Comment (atmorling) [inline ecoscope_workflows/tasks/results/_ecoplot.py:121]:
```suggestion
        BarLayoutStyle | SkipJsonSchema[None],
```
Updating here per above suggestion

**Review by atmorling — COMMENTED** [review]
Looks good to me. I've thrown some extra style params as a suggestion, but I realise this implies updates to the dag params/tests. Happy to punt those to a future PR for the sake of merging this one.

File: ecoscope_workflows/tasks/results/_ecomap.py (hunk 1 above)
Comment (cisaacstern) [reply ecoscope_workflows/tasks/results/_ecomap.py:48]: Opened https://github.com/wildlife-dynamics/ecoscope-workflows/issues/183 here to track upstream issue. Marking this as resolved pending upstream fix.

**Comment by cisaacstern** [comment]
> Looks good to me. I've thrown some extra style params as a suggestion, but I realise this implies updates to the dag params/tests. Happy to punt those to a future PR for the sake of merging this one.

@atmorling These look great, but yes let's address in a separate PR to expedite merging this. I've opened #184 and assigned this to you, to track this.

File: ecoscope_workflows/jsonschema.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from inspect import signature
-from typing import Any, get_args
+from typing import Any, get_args, Optional
```
Comment (cisaacstern) [inline ecoscope_workflows/jsonschema.py:2]:
```suggestion
from typing import Any, get_args
```

File: ecoscope_workflows/jsonschema.py [hunk 5]
```diff
@@ -92,3 +92,4 @@ def ser_model(self) -> dict[str, Any]:
 class ReactJSONSchemaFormConfiguration(BaseModel):
     title: str = "Ecoscope Workflow Configurations Form"
     properties: dict[str, Any]
+    definitions: Optional[dict[str, Any]] = Field(alias="$defs", default=None)
```
Comment (cisaacstern) [inline ecoscope_workflows/jsonschema.py:95]:
```suggestion
    definitions: dict[str, Any] | None = Field(alias="$defs", default=None)
```

File: ecoscope_workflows/tasks/results/_ecoplot.py [hunk 6]
```diff
@@ -43,7 +68,7 @@ def draw_ecoplot(
         grouped=grouped,
         x_col=x_axis,
         y_col=y_axis,
-        **style_kws,
+        **(plot_style.model_dump(exclude_none=True) if plot_style else {}),
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/results/_ecoplot.py:71]: Nice!

**Review by cisaacstern — COMMENTED** [review]
Awesome, Yun! Thanks for this really thoughtful diff.

I just have nit below to prefer `| None` syntax instead of `Optional`, which I'll merge myself since I assume it's a trivial stylistic change, and assuming tests pass I'll merge after that.

🙏

**Pull request merged**